[#161] 게시물: 게시물 화면 구현

jwonyLee opened the pull request:
## 개요

<img src="https://user-images.githubusercontent.com/15073405/141646623-9af5b672-e1c0-4ead-b9e0-9be4d66d15b1.gif" width="50%" />

글쓰기 화면 구현

## 작업 사항

- 게시물 화면 구현
  - 테이블 뷰, PostTableCell 를 이용하여 게시물 화면을 구현했습니다. 
- Style Guide 적용
  - backgroundColor 변경
  - commentMoreButton.color 변경
- DropDownView 구현
  - 우측 상단에 `moreMenuButton`을 탭했을 때 수정/삭제 메뉴를 구현했습니다. 현수님이 구현하신 `DropBoxView`를 많이 참고했습니다.
- 레이아웃 오류 수정
  - HorizontalScrollLabel 오토레이아웃 수정
  - PostTableCell.contentTextView 오토레이아웃 수정

## 예외 사항

- 실제 데이터 바인딩X
- `moreMenuButton` 수정/삭제 기능 동작X
- 좋아요, 댓글, 포토카드 버튼 동작X
  - 차례대로 탭했을 때 휴지통 화면, 사고싶다 화면, 기본 화면일 때의 모습을 나타내기 위한 하위 뷰 숨김/표시 처리하는 기능을 수행하고 있습니다.

동작은 설정 화면에서 확인할 수 있습니다. 혹시나 누락된 부분이 있다면 알려주시면 감사하겠습니다. 🙏

### Linked Issue

close #161

jwonyLee committed:
[#161] feat: 게시물 화면 구현
- 설정 화면을 통해 게시물 화면 표시
- TableView, PostTableCell로 기본 화면 구현

jwonyLee committed:
[#161] view: Style Guide 적용
- backgroundColor = primaryBackground
- commentMoreButton.color = gray2

jwonyLee committed:
[#161] view: DropDownView 구현

jwonyLee committed:
[#161] feat: DropDownView 외부를 터치했을 때 DropDownView가 닫히는 기능 구현
- DropBoxView 참고

jwonyLee committed:
[#161] fix: 레이아웃 오류 수정

gustn3965 commented on ThingLog/ViewController/Post/PostViewController+setup.swift:
```diff
@@ -0,0 +1,26 @@
+            tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
```
`bottomAnchor`는 `safe`보다는 `view.bottomAnchor`이 더 적절할 것 같은데 맞나요??

gustn3965 commented on ThingLog/View/DropDownView/DropDownView.swift:
```diff
@@ -0,0 +1,170 @@
+        ])
+    }
+
+    private func setupTableView() {
+        tableViewHeight = CGFloat(items.count * 44)
+        tableViewHeightConstraint = tableView.heightAnchor.constraint(equalToConstant: 0)
+        tableViewHeightConstraint?.isActive = true
+
```
`tableView`의 백그라운드 색도 지정해야 흰색또는 검은색으로 나타나지 않습니당 ~  
( 드롭박스 나타날때 딱 흰색 또는 검은색 바탕화면이 살짝 보여요 ) 
```swift
tableView.backgroundColor = SwiftGenColors.primaryBackground.color
```

gustn3965 commented on ThingLog/View/DropDownView/DropDownView.swift:
```diff
@@ -0,0 +1,170 @@
+    private func setupView() {
+        addSubviews(button)
+        superView?.addSubview(tableView)
+        bringSubviewToFront(tableView)
```
이 코드는 필요없을 것 같아요. `DropBoxView`에서도 동일하게 이 코드가 쓰여있는데, 아무래도 `superView?`가 빠진 듯 한데, 없어도 무방하다고 생각합니다. ( 테이블뷰를 보여줄 때 `superView`에서 다시 `bringSubviewToFront`를 적용하므로 )  코드를 잘못적었습니다. 죄송합니다! 🙏🏻   변경하실 때 `DropBoxView`의 사용된 저 코드도 같이 없애주시면 감사하겠습니다!

gustn3965 reviewed (approved):
LGTM! 고생하셨습니다.

코멘트들 확인하시고 직접 merge하시면 될 것 같아요!

gustn3965 commented:
추가적으로, 삭제/복구 버튼의 하단 `bottomLineView` 의 색이 다른 것 같아요~
아무래도` gray4` 가 아닌, `gray2` 일 것 같습니당. 

![image](https://user-images.githubusercontent.com/48749182/141683241-0d656d41-e7d9-4785-958f-acc75f2dcdd0.png)

jwonyLee committed:
[#161] ref: 피드백 반영
- PostViewController.tableView 오토레이아웃 수정
- DropDownView.tableView.backgroundColor 지정
- DropDownView, DropBoxView bringSubviewToFront(tableView) 제거
- LeftRightButtonView bottomLineView 색상 변경

jwonyLee committed:
Merge branch 'develop' into view/161-postVC

jwonyLee commented:
피드백 주신 부분 전부 반영했습니다~
